GoogleForCreators/web-stories-wp
# Improve save-blocking media uploading checks

## Context



Right now we disable saving a story if any upload is in progress, even if just uploading to the library and not the canvas

Instead, we only want to disable saving a story if any media item within the story itself is currently uploading, as only those are concerning due to blob URLs.

## Summary



This PR improves the media uploading checks we have in multiple places to only disable save buttons when any story elements are currently being uploaded.

So if I upload a video only to the media library, I can still save the story even when the upload is still in progress.

## Relevant Technical Choices



Uses a new hook for easier reusability

## To-do



* [x] Test that it works as expected with video poster generation

## User-facing changes



## Testing Instructions






This PR can be tested by following these steps:

1. Drag & drop a video or image to the media library
2. See that the story can be saved/published
3. Drag & drop a video or image to the canvas
4. See that the update/publish button is disabled while the upload is in progress


## Reviews

### Does this PR have a security-related impact?



No

### Does this PR change what data or activity we track or use?



No

### Does this PR have a legal-related impact?



No

## Checklist




---



Fixes #10341
